libraries/node-core-library/src/FileSystem.ts

```diff
@@ -785,6 +791,59 @@ export class FileSystem {
     });
   }
 
+  /**
+   * Writes the contents of multiple Uint8Arrays to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writevSync()`.
+   * This API is useful for writing large files efficiently, especially if the input is being concatenated from
+   * multiple sources.
+   * @remarks
+   * Throws an error if the folder doesn't exist, unless ensureFolder=true.
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The content that should be written to the file.
+   * @param options - Optional settings that can change the behavior.
+   */
+  public static writeBuffersToFile(
+    filePath: string,
+    contents: Iterable<Uint8Array>,
+    options?: IFileSystemWriteBinaryFileOptions
+  ): void {
+    FileSystem._wrapException(() => {
+      const toCopy: Uint8Array[] = [...contents];
```

Comment:
This seems to defeat the purpose of accepting `Iterable<T>`.  Isn't the whole point to avoid allocating the entire file on the heap at once?

Reply:
When I wrote this, I didn't realize that `filev()` itself accepts an array of buffers.

However I'm still uncomfortable with `[...contents]` because a caller who provides an efficient generator might be alarmed to discover that the full file gets allocated in memory. It might be better to accept a chunk size, e.g. if the size is 1MB then up to 1MB of buffers will be iterated, then written as a batch, then more buffers will be iterated.  Since you already have a `while` loop, this might not require much additional logic.

Reply:
Switched to accepting `ReadonlyArray<Uint8Array>`. Normally I like accepting `Iterable<T>` for things because it allows you to pass a `Set<T>` and does not require use of `T[]`.